quill-core/src/main/scala/io/getquill/ast/Ast.scala

```diff
@@ -111,3 +111,11 @@ case class Assignment(input: Ident, property: String, value: Ast)
 //************************************************************
 
 case class Dynamic(tree: Any) extends Ast
+
+case class Binding(key: String, code: String) extends Ast
+
+class CompileTimeBinding[T](override val key: String, val tree: T) extends Binding(key, tree.toString)
```

Comment:
`case class` inheritance is discouraged, maybe the hierarchy could be `trait Binding`, `case class RuntimeBinding`, and `case class CompiletimeBinding`?